pkg/web/db-actions.ts

```diff
@@ -0,0 +1,78 @@
+let database: Database = {
+  notes: [],
+};
+interface Database {
+  notes: Note[];
+}
+export interface Note {
+  _id: string;
+  books: BookInterface[];
+}
+export interface BookInterface {
+  isbn: string;
+  title: string;
+  author: string;
+  editor: string;
+  price: number;
+  quantity: number;
+  note?: string;
+  date?: string;
+}
+
+export const newNote = (note: Note) => {
+  const currentDatabase = getDB();
+
+  const newDB = { notes: [...currentDatabase.notes, note] };
+
+  setDB(newDB);
+
+  const promise = new Promise<{ id: string; ok: boolean }>((resolve) => {
+    resolve({ id: note._id, ok: true });
+  });
+  return promise;
+};
+
+// get note
+export const getNote = (id: Note["_id"]) => {
+  const currentDatabase = getDB();
+
+  const note =
+    currentDatabase.notes.find((note) => note._id === id) || ({} as Note);
+
+  const promise = new Promise<Note>((resolve) => {
+    resolve(note);
+  });
+
+  return promise;
+};
+
+// set DB
```

Comment:
Instead of having a "currently active" DB we could have a function `createDB` that returns an empty database.
The returned object would then be needed to invoke `newNote`, `updateNote` etc.